consensus/src/pipeline/buffer_manager.rs

```diff
@@ -882,11 +888,26 @@ impl BufferManager {
             while let Some(commit_msg) = commit_msg_rx.next().await {
                 let tx = verified_commit_msg_tx.clone();
                 let epoch_state_clone = epoch_state.clone();
+                let optimistic_sig_verification_for_commit_votes =
+                    self.optimistic_sig_verification_for_commit_votes;
                 bounded_executor
                     .spawn(async move {
+                        // When this flag is enabled, we skip verification of commit vote for now.
+                        // Once enough commit votes are received, we will aggregate and verifyt them optimistically.
+                        if optimistic_sig_verification_for_commit_votes {
+                            if let CommitMessage::Vote(vote) = &commit_msg.req {
+                                if !epoch_state_clone
+                                    .verifier
+                                    .is_malicious_author(&vote.author())
+                                {
+                                    let _ = tx.unbounded_send((commit_msg, false));
+                                    return;
+                                }
+                            }
+                        }
                         match commit_msg.req.verify(&epoch_state_clone.verifier) {
                             Ok(_) => {
-                                let _ = tx.unbounded_send(commit_msg);
+                                let _ = tx.unbounded_send((commit_msg, true));
```

Comment:
nit: Instead of `true` and `false`, it reads better if you can create an enum for `verified` and `unverified` and pass them.

Reply:
Done.